### PR TITLE
devices: add tests for gr716-multi

### PR DESCRIPTION
**Pull request by lukileczo**
JIRA: RTOS-534

## Description

- add `devices` directory for `phoenix-rtos-devices` tests
- add tests for `gr716-multi` driver
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: Makefile.sparcv8leon3 [hunk 1]
```diff
@@ -0,0 +1 @@
+DEFAULT_COMPONENTS += test-gr716-multi
```
Comment (agkaminski) [inline Makefile.sparcv8leon3:1]: No new line at the end of the file

**Review by agkaminski — COMMENTED** [review]
nitpick, otherwise LGTM

**Comment by agkaminski** [comment]
@damianloew could you pls review this too (or delegate someone else to review it instead)?

**Comment by damianloew** [comment]
> @damianloew could you pls review this too (or delegate someone else to review it instead)?



> @damianloew could you pls review this too (or delegate someone else to review it instead)?

Yes, sure.

File: devices/gr716-multi/Makefile [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+# GR716 multi tests
+#
+# Copyright 2023 Phoenix Systems
+#
+
+NAME := test-gr716-multi
+LOCAL_SRCS := gr716-multi.c
+LIBS := unity
+
+include $(binary.mk)
```
Comment (adamdebek) [inline devices/gr716-multi/Makefile:13]: LGTM, I would only change source file name from gr716-multi.c to test-gr716-mulit and use add_test macro:
```
$(eval $(call add_test, test-gr716-multi, unity))
```

**Review by adamdebek — APPROVED** [review]

**Pull request merged**